spinn_front_end_common/interface/java_caller.py

```diff
@@ -0,0 +1,158 @@
+from collections import OrderedDict
+import json
+import os
+import subprocess
+
+from pacman.exceptions import PacmanExternalAlgorithmFailedToCompleteException
+from pacman.utilities.file_format_converters.convert_to_java_machine import \
+    ConvertToJavaMachine
+from spinn_front_end_common.utilities.exceptions import ConfigurationException
+
+
+class JavaCaller(object):
+    """ Support class that holds all the stuff for running stuff in Java.
+
+        This includes the work of preparing data for transmitting to Java and
+        back.
+
+        This seperates the choices of how to call the Java batch vs streaming,
+            jar locations, paramters ect from the rest of the python code.
+    """
+
+    __slots__ = [
+        # The sqllite databse
+        "_database_file",
+        # The call to get java to work. Including the path if required.
+        "_java_call",
+        # The local https://github.com/SpiNNakerManchester/JavaSpiNNaker
+        "_javaspinnaker_path",
+        # the folder to write the any json files into
+        "_json_folder",
+        # The location where the machine json is written
+        "_machine_json",
+        # The location where the latest placement json is written
+        "_placement_json"
+    ]
+
+    def __init__(self, json_folder, java_call, javaspinnaker_path=None):
+        """
+        Creates a Jason caller and checks the user/ config parameters
+
+        :param json_folder: The location where the machine json is written.
+        :type json_folder: str
+        :param java_call: Call to start java. Including the path if required.
+        :type java_call: str
+        :param javaspinnaker_path: the path where the java code can be found.
+            This must point to a local copy of \
+            https://github.com/SpiNNakerManchester/JavaSpiNNaker. \
+            It must also have been built! \
+            If None the assumption is that it is the same parent directory as \
+            https://github.com/SpiNNakerManchester/SpiNNFrontEndCommon.
+        :raise ConfigurationException if simple parameter checking fails.
+        """
+        self._json_folder = json_folder
+
+        self._java_call = java_call
+        result = subprocess.call([self._java_call, '-version'])
+        if result != 0:
+            raise ConfigurationException(
+                " {} -version failed. "
+                "Please set [Java] java_call to the absolute path "
+                "to start java. (in config file)".format(self._java_call))
+
+        if javaspinnaker_path is None:
+            interface = os.path.dirname(os.path.realpath(__file__))
+            spinn_front_end_common = os.path.dirname(interface)
+            spinnfrontendcommon = os.path.dirname(spinn_front_end_common)
+            parent = os.path.dirname(spinnfrontendcommon)
+            self._javaspinnaker_path = os.path.join(parent, "JavaSpiNNaker")
+        else:
+            self._javaspinnaker_path = javaspinnaker_path
```

Comment:
```suggestion
            self._java_spinnaker_path = java_spinnaker_path
```

Reply:
if changing that makes you happy...

Reply:
it does. it makes it read easier. and stops pycharm complaining about typos